src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, {
+    x = 0, y = 0,
+  }, chipVersion) {
```

Comment:
you should use coords object to get x and y
```suggestion
  constructor(name, weight, coords = {x: 0, y: 0}, chipVersion) {
```